consensus/istanbul/backend/announce.md

```diff
@@ -0,0 +1,217 @@
+# Celo Announce Protocol
+
+While here referred as a protocol by itself, Announce is technically a subset of the Istanbul protocol, on the [RLPx] transport, that facilitates Validators to privately discover the `eNodeURL` of other Validators in the network, using the underlying p2p eth network. This is used by Istanbul to have direct connections between Validators, speeding up the consensus phase of block generation.
+
+It is important to note that a validator's `eNodeURL` is never shared among third parties, but directly published to specific peers. Therefore it is allowed for a validator to publish different external facing `eNodeURL` values to different validator peers.
+
+## Terminology
+
+For the purpose of this specification, certain terms are used that have a specific meaning, those are:
+
+* Full node
+* Validator
+* Nearly Elected Validator
+* eNodeURL
+
+### Full Node
+
+A Full Node is a Celo node fully operating in the Celo p2p network.
+
+### Validator
+
+A Validator is a [FullNode] that has been started with the ability to produce blocks, even if it has not yet been elected. In the geth program,
+this is equivalent as being started with the `--mine` param.
+
+### Nearly Elected Validator (NEV)
+
+A node is a Nearly Elected Validator (NEV) if and only if:
+
+1) It is a [Validator]
+2) It is in the result set from calling `ElectNValidatorSigners` with `additionalAboveMaxElectable = 10`
+
+In loose terms, it means it's a validator that has a good chance of becoming an elected validator in the following epoch.
+
+### eNodeURL
+
+An `eNodeURL` is a string specifyng the location of an eNode in the p2p network (//TODO add link to what an enode is?), and it has the following format:
+
+`enode://<hex node id>@<IP/hostname>:<tcp port>?discport=<udp discovery port>`
+
+Where `<hex node id>` is the node's serialized and hex encoded ecdsa public key. The URL parameter `discport` should only be specified if the udp discovery port differs from the tcp port.
+
+Some example `eNodeURLs` (with partially elided hex encoded public keys):
+
+```
+enode://517318.......607ff3@127.0.0.1:34055
+
+enode://e5e2fdf.......348ce8@127.0.0.1:33503?discport=22042
+```
+
+## Objective
+
+The Announce protocol objective is to allow all [NearlyElectedValidator] to maintain an updated table of `validator -> eNodeURL` for all other [NearlyElectedValidator], while somewhat concealing these values from the rest of the nodes in the network, and also avoiding an unnecessary high message traffic to achieve it. [NearlyElectedValidator] nodes can use the information from this table to open direct connections between themselves, speeding up the consensus phase.
```

Comment:
This seems to be missing a reference to the currently elected validators. Also I think maybe changing `validator -> eNodeURL` to `validatorAddress -> eNodeURL` makes things a little clearer.